### PR TITLE
ngc: implement service

### LINKED ISSUE #11494: Baten Kaitos HD hangs on loading screen after booting
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

EA 3867

### Description of Issue

There's an in-game selection between Baten Kaitos 1 and 2. After selecting either of them, yuzu shows a launching screen as if it's starting a new game, then shows a black screen instead of what should be the loading screen for BK.

Selection screen: 0100f28018ca4000_2023-09-13_23-40-21-838
Loading screen that hangs indefinitely: 0100f28018ca4001_2023-09-13_23-40-27-511

Note: The loading screen in the second screenshot is completely black (no bar visible) in the emulator. It only shows up in the screenshot.

### Expected Behavior

Selecting BK1 or 2 should show a loading screen, which eventually ends in launching either game.

### Reproduction Steps

1) Start game
2) Wait until selection screen for BK1 or 2
3) Select one of the games and confirm

### Log File

[yuzu_log (5).txt](https://github.com/yuzu-emu/yuzu/files/12602193/yuzu_log.5.txt)
[yuzu_log (4).txt](https://github.com/yuzu-emu/yuzu/files/12602195/yuzu_log.4.txt)

### System Configuration

Steam Deck

### PR DESCRIPTION
**Pull request by liamwhite**
This was added in 16.0.0. Possibly fixes #11494

### CODE REVIEW

File: src/core/hle/service/ngc/ngc.cpp [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+private:
+    static constexpr u32 NgcContentVersion = 1;
+
+    // This is nn::ngc::detail::ProfanityFilterOption
+    struct ProfanityFilterOption {
+        INSERT_PADDING_BYTES_NOINIT(0x20);
+    };
+    static_assert(sizeof(ProfanityFilterOption) == 0x20);
```
Comment (FearlessTobi) [inline src/core/hle/service/ngc/ngc.cpp:77]: should have a string like "xyz has incorrect size" as its second argument

**Comment by brujo5** [comment]
Screenshot_20230914-190145
Screenshot_20230914-190252_yuzu

I just tried the Android v71 version where it already includes this code and the game remains the same, black screen after selecting any of the games

Log file 
[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/12614898/yuzu_log.txt)

**Comment by liamwhite** [comment]
Multi-program applications are not implemented for Android yet

**Comment by brujo5** [comment]
Sadly, we will have to wait then.Thank you for your response

**Pull request merged**